app/main.py

```diff
@@ -1,5 +1,81 @@
+from typing import List
+
+
 class Cargo:
-    def __init__(self, weight: int) -> None:
+    def __init__(
+            self,
+            weight: int
+    ) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot(Cargo):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: List[int] = None
+    ) -> None:
+        super().__init__(weight)
+        self.name = name
+        self.coords = coords if coords is not None else [0, 0]
+
+    def go_forward(
+            self,
+            step: int = 1
+    ) -> None:
+        self.coords[1] += 1 * step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= 1 * step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += 1 * step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= 1 * step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: List[int] = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+
+        if self.coords is None or len(self.coords) == 2:
+            self.coords = [0, 0, 0]
```

Comment:
There is no need to redefine the `coords` after initialization in the parent class, pass to the `super().__init__` the right value from the beginning.
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```